test/unittests/skills/test_context.py

```diff
@@ -0,0 +1,42 @@
+from unittest import TestCase, mock
+
+from mycroft.skills.context import adds_context, removes_context
+"""
+Tests for the adapt context decorators.
+"""
+
+
+class ContextSkillMock(mock.Mock):
+    """Mock class to apply decorators on."""
+    @adds_context('DestroyContext')
+    def handler_adding_context(self):
+        pass
+
+    @adds_context('DestroyContext', 'exterminate')
+    def handler_adding_context_with_words(self):
+        pass
+
+    @removes_context('DestroyContext')
+    def handler_removing_context(self):
+        pass
+
+
+class TestContextDecorators(TestCase):
+    def test_adding_context(self):
+        """Check that calling handler adds the correct Keyword."""
+        skill = ContextSkillMock()
+        skill.handler_adding_context()
+        skill.set_context.assert_called_once_with('DestroyContext', '')
+
+    def test_adding_context_with_words(self):
```

Comment:
Is it worth one more test - `test_getting_context` - getting the added context and testing that it returns the optional string
Or do you want to keep this solely focused on the decorators being called correctly?

Reply:
I have that started in a separate PR I'm working on (Adding tests and refactoring the intent service) but I can include that in this PR as well, it will be placed in the test_intent_service.py file and then moved in my follow up PR when it's done.

Reply:
And there is actually partial test coverage for that already in the test_intent_service.py

Reply:
Don't want to create double handling. This tests the decorators get called correctly, great to hear of new tests on the way!